user.go

```diff
@@ -203,6 +210,10 @@ func (u User) Equal(x User) bool {
 		return false
 	}
 
+	if !reflect.DeepEqual(u.MfaFactors, x.MfaFactors) {
+		return false
+	}
```

Comment:
While `reflect` works, it is generally considered to be very slow and where possible we prefer manual checking for performance reasons.

You can manually check these maps are equal using the following:

```go
	if len(u.MFAFactors) != len(x.MFAFactors) {
		return false
	}
	for k, v := range u.MFAFactors {
		if xV, ok := x.MFAFactors[k]; !ok || xV != v {
			return false
		}
	}
```